gdal/ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp

```diff
@@ -2108,19 +2181,43 @@ bool OGRGeoJSONBaseReader::GenerateFeatureDefn( OGRLayer* poLayer,
                                                "TRUE"));
                     if( bFlattenGeocouchSpatiallistFormat )
                     {
-                        poDefn->DeleteFieldDefn(poDefn->GetFieldIndexCaseSensitive("type"));
+                        auto typeIter = oMapFieldNameToIdx.find("type");
+                        if( typeIter != oMapFieldNameToIdx.end() )
+                        {
+                            const int nIdx = typeIter->second;
+                            apoFieldDefn.erase(apoFieldDefn.begin() + nIdx);
+                            oMapFieldNameToIdx.erase(typeIter);
+                            dag.removeNode(nIdx);
+                        }
+
                         bIsGeocouchSpatiallistFormat = true;
-                        return GenerateFeatureDefn(poLayer, poObj);
+                        return GenerateFeatureDefn(oMapFieldNameToIdx,
+                                                   apoFieldDefn,
+                                                   dag,
+                                                   poLayer, poObj);
                     }
                 }
             }
 
-            OGRGeoJSONReaderAddOrUpdateField(poDefn, it.key, it.val,
+            anCurFieldIndices.clear();
+            OGRGeoJSONReaderAddOrUpdateField(anCurFieldIndices,
+                                             oMapFieldNameToIdx,
+                                             apoFieldDefn,
+                                             it.key, it.val,
                                              bFlattenNestedAttributes_,
                                              chNestedAttributeSeparator_,
                                              bArrayAsString_,
                                              bDateAsString_,
                                              aoSetUndeterminedTypeFields_);
+            for( int idx: anCurFieldIndices )
+            {
+                dag.addNode(idx, apoFieldDefn[idx]->GetNameRef());
+                if( nPrevFieldIdx != -1 )
```

Comment:
if you ever decide to do edge-counting, the n^2 loop with all pairs and not just adjacent neighbors will get more stable voting.

Reply:
I've added a commit with a comment in the code pointing to that PR if we need those enhancements